lib/base/utility.cpp

```diff
@@ -1265,6 +1266,25 @@ int Utility::CompareVersion(const String& v1, const String& v2)
 	return 0;
 }
 
+String Utility::ParseVersion(const String& version)
+{
+	bool ret;
+	boost::smatch match;
+
+	try {
+		boost::regex expr("[rv]?(\\d+.\\d+.\\d+).*");
+		String operand = version;
+		ret = boost::regex_search(operand.GetData(), match, expr);
+	} catch (boost::exception&) {
+		ret = false;
+	}
+
+	if (ret)
+		return std::string(match[1].first, match[1].second);
```

Comment:
The `String` class has a ctor which takes two iterators (template<typename InputIterator> String(InputIterator begin, InputIterator end)) which should be preferred over std::string.

Reply:
Did not work in my first tests, now it does. Thanks.